pagefaultgames/pokerogue
# Enhance Gameplay Reliability with Offline Mode, Automatic Reconnection, and Save File Synchronization

## What are the changes?
Previously, losing connection to the server interrupted gameplay. With these changes:
- Players can continue offline if the server is down.
- The game periodically checks server status and reconnects automatically.
- Reconnection is smooth, without needing a page reload.
- A banner shows the current server status.
- Players can refresh the page and still play offline if they connected at least once.
- Save files are synchronized with the server when it becomes available again.

## Why am I making these changes?
These improvements aim to provide an uninterrupted gaming experience, even during server downtimes or DDoS attacks. By enabling offline play, smooth reconnections, and save file synchronization, we ensure players can continue their game without disruptions, enhancing overall user satisfaction.

## How to test the changes?
To test:
1. Disconnect from the internet and start the game to verify offline functionality.
2. Reconnect to the internet and check for automatic server reconnection without page reload.
3. Verify the banner shows server status.
4. Refresh the page after connecting once and confirm offline functionality.
5. Ensure save files synchronize with the server when it comes back online.

## Checklist
- [x] Have I checked for no overlap with another PR?
- [x] Is the PR self-contained and not splittable into smaller PRs?
- [x] Have I clearly explained the changes?
- [x] Are there visual changes?
